ckan/lib/helpers.py

```diff
@@ -868,7 +868,11 @@ def dict_list_reduce(list_, key, unique=True):
     values for the key with unique values if requested. '''
     new_list = []
     for item in list_:
-        value = item.get(key)
+        try:
+            value = item.get(key)
+        except AttributeError:
+            # item might not be a dict.
+            continue
```

Comment:
It appears that the `package.resources` object passed to `dict_list_reduce` is actually not a `list` but a `dict` (corresponding to one resource). But the package at stake actually has two resources.